packages/backend-function/src/provided_function_factory.ts

```diff
@@ -0,0 +1,143 @@
+  BackendOutputStorageStrategy,
+  ConstructContainerEntryGenerator,
+  ConstructFactory,
+  ConstructFactoryGetInstanceProps,
+  FunctionResources,
+  GenerateContainerEntryProps,
+  ResourceAccessAcceptor,
+} from '@aws-amplify/plugin-types';
+import { Construct } from 'constructs';
+import { CfnFunction, IFunction } from 'aws-cdk-lib/aws-lambda';
+import { FunctionOutput } from '@aws-amplify/backend-output-schemas';
+import { Tags } from 'aws-cdk-lib';
+import { AmplifyUserError, TagName } from '@aws-amplify/platform-core';
+import { AmplifyFunctionBase } from './function_construct_base.js';
+import { FunctionResourceAccessAcceptor } from './resource_access_acceptor.js';
+
+export type ProvidedFunctionProps = {
+  /**
+   * Group the function with existing Amplify resources or separate the function into its own group.
+   * @default 'function' // grouping with other Amplify functions
+   * @example
+   * resourceGroupName: 'auth' // to group an auth trigger with an auth resource
+   */
+  resourceGroupName?: AmplifyResourceGroupName;
+};
+
+/**
+ * Adapts provided CDK function as Amplify function.
+ */
+export class ProvidedFunctionFactory
+  implements ConstructFactory<AmplifyFunction>
+{
+  private generator: ConstructContainerEntryGenerator;
+
+  /**
+   * Creates provided function factory.
+   */
+  constructor(
+    private readonly functionProvider: (scope: Construct) => IFunction,
+    private readonly props?: ProvidedFunctionProps
+  ) {}
+
+  /**
+   * Creates a function instance.
+   */
+  getInstance(props: ConstructFactoryGetInstanceProps): AmplifyFunction {
+    if (!this.generator) {
+      this.generator = new ProvidedFunctionGenerator(
+        this.functionProvider,
+        props.outputStorageStrategy,
+        this.props
+      );
+    }
+    return props.constructContainer.getOrCompute(
+      this.generator
+    ) as AmplifyFunction;
+  }
+}
+
+class ProvidedFunctionGenerator implements ConstructContainerEntryGenerator {
+  readonly resourceGroupName: AmplifyResourceGroupName;
+
+  constructor(
+    private readonly functionProvider: (scope: Construct) => IFunction,
+    private readonly outputStorageStrategy: BackendOutputStorageStrategy<FunctionOutput>,
+    props?: ProvidedFunctionProps
+  ) {
+    this.resourceGroupName = props?.resourceGroupName ?? 'function';
+  }
+
+  generateContainerEntry = ({ scope }: GenerateContainerEntryProps) => {
+    let providedFunction: IFunction;
+    try {
+      providedFunction = this.functionProvider(scope);
+    } catch (e) {
+      if (
+        e instanceof Error &&
+        (e.message.includes('docker exited with status 1') ||
+          e.message.includes('docker ENOENT'))
+      ) {
+        throw new AmplifyUserError(
+          'FunctionBundlingDockerError',
+          {
+            message: e.message,
+            // TODO better resolution
+            resolution:
+              'Ensure that docker is present and works correctly. See https://some.link.about.docker.',
```

Comment:
```suggestion
        throw new AmplifyUserError(
          'CustomFunctionProviderDockerRequiredError',
          {
            message: e.message,
            resolution:
              'See https://docs.amplify.aws/react/build-a-backend/functions/custom-functions for more details about current limitations and troubleshooting steps.',
```

@josefaidt @Amplifiyer please take a look at proposed wording here. The goal here is to:
- refrain from stating docker support in error message/resolution in case we find a way to support it (avoid hardcoding).
- point to web page that we can edit out of bound with most recent information about the state of support. (providing support should not involve code change in this repo, therefore need for decoupling).

Reply:
We have that callout at the top of this page that calls out limitations with docker, but perhaps this warrants more clarity at the top of that page -- prereqs, etc.

There's a note I left on the docs PR to (separately) create a troubleshooting dedicated to docker errors which we can use to add more color for support